volmdlr/edges.py

```diff
@@ -2131,29 +2151,76 @@ def cut_between_two_points(self, point1, point2):
         return arc
 
     def line_intersections(self, line2d: Line2D, tol=1e-9):
-        # Duplicate from circle
-        Q = self.center
-        if line2d.points[0] == self.center:
-            P1 = line2d.points[1]
-            V = line2d.points[0] - line2d.points[1]
+        try:
+            if line2d.start == self.center:
+                pt1 = line2d.end
+                vec = line2d.start - line2d.end
+            else:
+                pt1 = line2d.start
+                vec = line2d.end - line2d.start
+        except AttributeError:
+            if line2d.point1 == self.center:
+                pt1 = line2d.point2
+                vec = line2d.point1 - line2d.point2
+            else:
+                pt1 = line2d.point1
+                vec = line2d.point2 - line2d.point1
+        a = vec.dot(vec)
+        b = 2 * vec.dot(pt1 - self.center)
+        c = pt1.dot(pt1) + self.center.dot(self.center) \
+            - 2 * pt1.dot(self.center) - self.radius ** 2
+
+        disc = b ** 2 - 4 * a * c
+        if math.isclose(disc, 0., abs_tol=tol):
+            t1 = -b / (2 * a)
+            return [pt1 + t1 * vec]
+
+        elif disc > 0:
+            sqrt_disc = math.sqrt(disc)
+            t1 = (-b + sqrt_disc) / (2 * a)
+            t2 = (-b - sqrt_disc) / (2 * a)
+            return [pt1 + t1 * vec,
+                    pt1 + t2 * vec]
         else:
-            P1 = line2d.points[0]
-            V = line2d.points[1] - line2d.points[0]
-        a = V.dot(V)
-        b = 2 * V.dot(P1 - Q)
-        c = P1.dot(P1) + Q.dot(Q) - 2 * P1.dot(Q) - self.radius ** 2
+            return []
+
```

Comment:
The last else statement may be removed

Reply:
Done in https://github.com/Dessia-tech/volmdlr/pull/405/commits/f257d57f5c14af60dcfb425b07ca0e4d5b95b504